fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -2678,4 +2681,85 @@ public void testUpdateRecords() {
             recordStore.saveRecord(evolvedAgainRecord);
         }
     }
+
+    @Nonnull
+    private FDBMetaDataStore openMetaDataStore(@Nonnull FDBRecordContext context, @Nonnull KeySpacePath metaDataPath, boolean clear) {
+        FDBMetaDataStore metaDataStore = new FDBMetaDataStore(context, metaDataPath);
+        metaDataStore.setDependencies(new Descriptors.FileDescriptor[] {
+                RecordMetaDataOptionsProto.getDescriptor()
+        });
+        metaDataStore.setMaintainHistory(false);
+        if (clear) {
+            context.ensureActive().clear(Range.startsWith(metaDataPath.toSubspace(context).pack()));
+        }
+        return metaDataStore;
+    }
+
+    @Test
+    public void testNoUnion() {
+        final KeySpacePath metaDataPath = TestKeySpace.getKeyspacePath(new Object[]{"record-test", "unit", "metadataStore"});
+        int version;
+        try (FDBRecordContext context = fdb.openContext()) {
+            FDBMetaDataStore metaDataStore = openMetaDataStore(context, metaDataPath, true);
+            metaDataStore.saveRecordMetaData(TestNoUnionProto.getDescriptor());
+            context.commit();
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("MySimpleRecord"));
+            version = metaDataStore.getRecordMetaData().getVersion();
+        }
+
+        // Store a MySimpleRecord record.
+        try (FDBRecordContext context = fdb.openContext()) {
+            FDBMetaDataStore metaDataStore = openMetaDataStore(context, metaDataPath, false);
+            FDBRecordStore recordStore = FDBRecordStore.newBuilder().setContext(context).setKeySpacePath(path)
+                    .setMetaDataStore(metaDataStore)
+                    .createOrOpen();
+            assertEquals(version, recordStore.getRecordMetaData().getVersion());
+            Descriptors.Descriptor mySimpleRecordDescriptor = recordStore.getRecordMetaData().getRecordType("MySimpleRecord").getDescriptor();
+            final Descriptors.FieldDescriptor recNo = mySimpleRecordDescriptor.findFieldByName("rec_no");
+            final Descriptors.FieldDescriptor numValue2 = mySimpleRecordDescriptor.findFieldByName("num_value_2");
+            final Descriptors.FieldDescriptor strValueIndexed = mySimpleRecordDescriptor.findFieldByName("str_value_indexed");
+            final Descriptors.FieldDescriptor numValue3Indexed = mySimpleRecordDescriptor.findFieldByName("num_value_3_indexed");
+            Message.Builder messageBuilder = DynamicMessage.newBuilder(mySimpleRecordDescriptor);
+            messageBuilder.setField(recNo, 1066L);
+            messageBuilder.setField(numValue2, 42);
+            messageBuilder.setField(strValueIndexed, "value");
+            messageBuilder.setField(numValue3Indexed, 1729);
+            recordStore.saveRecord(messageBuilder.build());
```

Comment:
That's right! I forgot to mention that in the commit description. On one hand, it doesn't feel right to me to assume a local file descriptor must always have a union. On the other hand, we cannot add the union on the fly, as it will change the descriptors and we will end up having the same problem we used to have before the introduction of local file descriptors (i.e., descriptors mismatch). I'm experimenting a little bit to see how I can work around this.

Reply:
I couldn't really find a good way around this, and if we don't find a way, I may take this PR back as I think it defeats the purpose of the PR; i.e., ease of use. @MMcM and I discussed this offline, for now lets not review this PR. I'll schedule a meeting to brainstorm possible solutions. Thanks.

Reply:
yeah, but the next test case uses local file descriptor.

Reply:
Oh, I see.